src/lib/engines/layerCompositor/fragment.glsl

```diff
@@ -21,6 +24,26 @@ void main() {
 
   vec4 layerColor = texture2D( layerMap, texelUv, mipmapBias );
 
+  if ( maskMode > 0 ) {
+    
+    vec2 maskTexelUv = ( uvToMaskTexture * vec3( v_uv, 1.0 ) ).xy;
+    vec4 maskTexel = texture2D( maskMap, maskTexelUv, mipmapBias );
+
+    float maskValue;
+
+    if( maskMode == 2 || maskMode == 4 ) {
+      maskValue = maskTexel.r;
+    } else {
+      maskValue = maskTexel.a;
+    }
+
+    if( maskMode >= 3 ) {
+      maskValue = 1.0 - maskValue;
+    }
+
+    layerColor *= maskValue;
```

Comment:
based on the lines directly below this, it looks like this isn't handling premultiplied alpha correctly in all cases

Reply:
Fixed. The premultiplication needed to happen at the input.